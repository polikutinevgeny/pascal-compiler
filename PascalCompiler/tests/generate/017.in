begin
    write(10 mod 3);
    write(#13);
    write(#10);
    write(10 div 3);
end.