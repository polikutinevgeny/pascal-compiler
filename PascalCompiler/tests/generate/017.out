.686p
include \masm32\include\masm32rt.inc
.mmx
.xmm
.code
main PROC
Push Ebp
Mov Ebp, Esp
Sub Esp, 0
Push 10
Push 3
Pop Ebx
Pop Eax
Cdq
Idiv Ebx
Push Edx
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 13
Mov Eax, Esp
printf("%c", dword ptr [Eax])
Sub Esp, 4
Push 10
Mov Eax, Esp
printf("%c", dword ptr [Eax])
Sub Esp, 4
Push 10
Push 3
Pop Ebx
Pop Eax
Cdq
Idiv Ebx
Push Eax
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Mov Esp, Ebp
Pop Ebp
exit
main ENDP
END main
